src/metrics.jl

```diff
@@ -222,7 +222,7 @@ Base.@propagate_inbounds function _evaluate(d::UnionMetrics, a::AbstractArray, b
     @inbounds begin
         s = eval_start(d, a, b)
         if IndexStyle(a, b) === IndexLinear() || size(a) == size(b)
-            @simd for I in 1:length(a)
+            @simd for I in eachindex(a, b)
```

Comment:
Note https://github.com/JuliaLang/julia/issues/28226#issuecomment-406977072.

Is that no longer the case?

Reply:
I tested https://github.com/JuliaStats/Distances.jl/pull/150#issuecomment-538310651, and obtained the results above. Wasn't this the test that would determine the slowdown?

Reply:
Sorry, I meant https://github.com/JuliaStats/Distances.jl/pull/107#issuecomment-473232505

Reply:
I get that `colwise` is consistently and measurably faster than `naivecolwise` in that test.

Reply:
I guess it is fixed then...

Reply:
I'll quickly double-check on Julia v1.0...

Reply:
Alright, I have tested my but off, and I can't seem to be able to reproduce the slowdown anymore, even on Julia v1.0. While the issue https://github.com/JuliaLang/julia/issues/28226 persists even on nightlies, the issue https://github.com/JuliaStats/Distances.jl/issues/100 does not re-occur with this PR, neither on Julia v1.0.5 nor v1.3 or nightlies. AFAICT, we wouldn't introduce a slowdown with the change in the line commented on above.